plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -798,13 +907,20 @@ private void writeEqualityDeleteToNationTable(Table icebergTable, Optional<Parti
 
     private void writeEqualityDeleteToNationTable(Table icebergTable, Optional<PartitionSpec> partitionSpec, Optional<PartitionData> partitionData, Map<String, Object> overwriteValues)
             throws Exception
+    {
+        writeEqualityDeleteToNationTableWithDeleteColumns(icebergTable, partitionSpec, partitionData, overwriteValues, Optional.empty());
+    }
+
+    private void writeEqualityDeleteToNationTableWithDeleteColumns(Table icebergTable, Optional<PartitionSpec> partitionSpec, Optional<PartitionData> partitionData, Map<String, Object> overwriteValues, Optional<List<String>> deleteFileColumns)
+            throws Exception
     {
         Path metadataDir = new Path(metastoreDir.toURI());
         String deleteFileName = "delete_file_" + UUID.randomUUID();
         FileIO fileIo = new ForwardingFileIo(fileSystemFactory.create(SESSION));
 
-        Schema deleteRowSchema = icebergTable.schema().select(overwriteValues.keySet());
-        List<Integer> equalityFieldIds = overwriteValues.keySet().stream()
+        List<String> deleteColumns = deleteFileColumns.orElse(new ArrayList<>(overwriteValues.keySet()));
+        Schema deleteRowSchema = icebergTable.schema().select(deleteColumns);
+        List<Integer> equalityFieldIds = deleteColumns.stream()
                 .map(name -> deleteRowSchema.findField(name).fieldId())
```

Comment:
We need pass equalityFieldIds too, otherwise we don't known nested schema's ids. `root.nested` schema's ids will be `root` or `root.nested`? @findinpath